screenwriter/screenwriter.go

```diff
@@ -0,0 +1,60 @@
+package screenwriter
+
+import (
+	"fmt"
+	"io"
+	"strings"
+)
+
+type ScreenWriter struct {
+	writer   io.StringWriter
+	previous string
+}
+
+func New(writer io.StringWriter) *ScreenWriter {
+	return &ScreenWriter{
+		writer: writer,
+	}
+}
+
+func (w *ScreenWriter) ClearPrevious() error {
+	if w.previous == "" {
+		return nil
+	}
+
+	// Get number of lines in previous
+	lines := strings.Count(w.previous, "\n")
+
+	// Handle the final line
+	finalLineLength := 0
+	splitLines := strings.Split(w.previous, "\n")
+	finalLine := splitLines[len(splitLines)-1]
+	finalLineLength = len(finalLine)
+	w.writer.WriteString(fmt.Sprintf("\u001b[%dD", finalLineLength))
```

Comment:
What does this line do?

Reply:
The escape sequence `\e[xD` moves the cursor back (i.e. to the left on a left-to-right display) by `x` columns, e.g. `\e[2D` moves it 2 columns to the left.

Reply:
Thanks, but it was more of a "please answer this for everyone" question :)

Reply:
Added a comment 👍